android/beagle/src/main/java/br/com/zup/beagle/android/context/operations/common/ExtractValueFromExpressionPDA.kt

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2020 ZUP IT SERVICOS EM TECNOLOGIA E INOVACAO SA
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package br.com.zup.beagle.android.context.operations.common
+
+import br.com.zup.beagle.android.context.operations.grammar.GrammarChars
+
+/**
+ * This class is able to read an Expression/Array input and output the value using PDA
+ *
+ * @param inputValue Input value is the string to be read by the Algorithm
+ *
+ * @param extractValueTypes The input method to algorithm know's the expression
+ *                          or array that must be read
+ *
+ * @see ExtractValueTypes.OPERATION Read the operation input e output the value
+ *
+ * @see ExtractValueTypes.ARRAY_PARAMETER Read the array value in a operation
+ *
+ * **/
+
+internal class ExtractValueFromExpressionPDA(
+    private val inputValue: String,
+    private val extractValueTypes: ExtractValueTypes
+) {
+
+    private var delimiterStartIndex: Int = 0
+    var startDelimiterChar: Char? = null
+    var endDelimiterChar: Char? = null
+
+    init {
+        if (extractValueTypes == ExtractValueTypes.OPERATION) {
+            startDelimiterChar = GrammarChars.OPEN_PARENTHESES
+            endDelimiterChar = GrammarChars.CLOSE_PARENTHESES
+        } else {
+            startDelimiterChar = GrammarChars.OPEN_BRACKET
+            endDelimiterChar = GrammarChars.CLOSE_BRACKET
```

Comment:
If I add more operations this code will be an increase, what do you think about create some think to get this logic?

Example of some other code to see an example:
```suggestion
            startDelimiterChar = extractValueTypes.operation.start
            endDelimiterChar = extractValueTypes.operation.end
```
You can create an enum with the variable `start` and `end` then you set the `GrammarChars`